### PR TITLE
Improve handling of missing endpoints/authentications

### PR DESCRIPTION
**Pull request by agrare**
If an endpoint or authentication is missing raise a helpful error rather
than blowing up on a no method for nilclass error.

Example:
```
RuntimeError (Unable to find a default endpoint for source [1])
```

### CODE REVIEW

File: lib/topological_inventory/openshift/operations/core/service_catalog_client.rb [hunk 1]
```diff
@@ -11,17 +11,13 @@ module Core
         class ServiceCatalogClient
           include Logging
 
-          attr_accessor :default_endpoint, :authentication, :connection_manager, :sleep_poll, :identity
+          attr_accessor :connection_manager, :sleep_poll, :source_id, :identity
+          attr_writer   :authentication, :default_endpoint
```
Comment (carbonin) [inline lib/topological_inventory/openshift/operations/core/service_catalog_client.rb]: Do these even need to be writers at this point?

File: lib/topological_inventory/openshift/operations/core/service_catalog_client.rb [hunk 2]
```diff
@@ -83,14 +87,33 @@ def service_instance_provisioning?(service_instance)
...
+          end
+
+          def fetch_authentication
+            endpoint = default_endpoint
+            return if endpoint.nil?
+
+            endpoint_authentications = sources_api_client.list_endpoint_authentications(endpoint.id.to_s).data || []
+            return if endpoint_authentications.empty?
+
+            auth_id = endpoint_authentications.first.id.to_s
```
Comment (carbonin) [inline lib/topological_inventory/openshift/operations/core/service_catalog_client.rb]: This `to_s` wasn't there before. It doesn't look needed: https://github.com/ManageIQ/topological_inventory-openshift/blob/master/lib/topological_inventory/openshift/operations/core/authentication_retriever.rb#L9

File: lib/topological_inventory/openshift/operations/core/service_catalog_client.rb (hunk 1 above)
Comment (agrare) [reply lib/topological_inventory/openshift/operations/core/service_catalog_client.rb]: Nope, removed

File: lib/topological_inventory/openshift/operations/core/service_catalog_client.rb (hunk 2 above)
Comment (agrare) [reply lib/topological_inventory/openshift/operations/core/service_catalog_client.rb]: Done

**Pull request merged**